src/lib/tls/msg_cert_verify.cpp

```diff
@@ -62,11 +62,11 @@ std::vector<uint8_t> Certificate_Verify::serialize() const
    {
    std::vector<uint8_t> buf;
 
-   if(m_scheme != Signature_Scheme::NONE)
+   // TODO: both for TLS 1.2 and 1.3 this should never happen and the message would be invalid
+   if(m_scheme.is_set())
```

Comment:
This check should not be necessary. Otherwise the message we create would be malformed.

Reply:
In this case should it be a `BOTAN_ASSERT` instead?